File: projects/engineering.purdue.edu/kak/scriptingwithobjects/swocode/chap13/GetKeySym.py
#!/usr/bin/env python

### GetKeySym.py

from Tkinter import * 

mw = Tk()                                                            #(A)
mw.title( "Testing Callback" )                                       #(B)       
mw.bind( '<KeyPress>', lambda e: get_key_info(e) )                   #(C)

def get_key_info(event):                                             #(D)
    (key_sym_name, key_num ) = (event.keysym, event.keysym_num)      #(E)
    print "keysym=%s, numeric=%s\n" % (key_sym_name, key_num)        #(F)

mainloop()                                                           #(G)

File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap13/GetKeySym.pl
#!/usr/bin/perl -w

### GetKeySym.pl

use strict;
use Tk;

my $mw = MainWindow->new( -title => "Testing Callback" );            #(A)       

$mw->bind( '<KeyPress>' => \&get_key_info );                         #(B)

sub get_key_info {                                                   #(C)
#    my $w = $Tk::widget;                                            #(D)
#    my $e = $w->XEvent;                                             #(E)
    my $e = $Tk::event;                                              #(F)
    my ($key_sym_name, $key_num ) = ($e->K, $e->N);                  #(G)
    print "keysym=$key_sym_name, numeric=$key_num\n";                #(H)
}

MainLoop;                                                            #(I)
